articles/nginx-reverse-proxy/index.md

```diff
@@ -0,0 +1,147 @@
+
+### Reverse Proxy
+A reverse proxy is a type of proxy server that retrieves resources on behalf of a client from one or more servers. These resources are then returned to the client, appearing as if they originated from the server itself.   
+
+### NGINX
+We will be using NGINX which would be serving the purpose of a Reverse Proxy. NGINX is a web server that can be used as a reverse proxy, load balancer, mail proxy and HTTP cache. 
+
+### Example
+![image](./reverse_proxy_working.png)
+
+This is an example of an architecture, where two apps are running in the background, but clients have no idea about them. Clients only know about NGINX which acts as a reverse proxy that sends the request to the appropriate application.
+
+
+Now that you have a broad idea of what we are about to build, let us jump right in!
+
+## Aim
+1. Deploy two applications and have it managed by NGINX.
+
+## Setup & Pre-Requisites:
+
+1. For this example, I have two sample NodeJS servers. One can have any kind of application running on different ports. 
+**NOTE:** Do not run your application on Port 80 or 443. Will be explained later why this must not be done.
+2. I have installed NGINX on my local machine, the same could be done on any Virtual Machine where the applications are expected to be deployed.
+
+## Step 1: Start two apps running in different ports
+As I've mentioned earlier, I got two Node Apps running on two different ports as shown below.  
+
+**Server app running on Port 3000**
+
+![image](./server1.png)  
+
+**Client app running on Port 3001**  
+
+![image](./client1.png)
+
+Now that we have our apps up and running, we don't want our users to use these applications by typing their PORTS explicitly, so we need to map it with something which is more human-readable. In this example, I will be using subdomains to distinguish between them. Again one is free to use whichever distinguishing element suitable as per their requirement.
+
+## Step 2: Add DNS records
+This is the part where one would add the DNS records in their DNS management dashboard. Since I am running it all locally I wouldn't be needing that.  
+
+The general DNS Configurations would be something like
+- Server app mapped to the server.domain
+- Client app mapped to the client.domain
+
+My Localhost Config, in this case, would be
+- Server mapped to server.localhost
+- Client mapped to client.localhost
+
+There are two standard protocols HTTP and HTTPS. The default port for HTTP is 80 and HTTPS is 443. This is the reason we must not run our applications on these ports because our NGINX server would be running on these two ports. All the requests the client makes would either be redirected to Port 80 or 443 from where it would be redirected internally to the corresponding application.
+
+## Step 4 - Configure NGINX at 80 for HTTP and 443 for HTTPS
+Now that we have our apps running and our DNS records ready. We can start configuring our NGINX to make it all work.
```

Comment:
Add reverse proxy after Nginx.

Reply:
Replaced "NGINX" to "NGINX Reverse Proxy"